docs/src/main/sphinx/connector/druid.rst

```diff
@@ -57,6 +57,73 @@ name from the properties file.
 Type mapping
 ------------
 
+Because Trino and Druid each support types that the other does not, this
+connector modifies some types when reading or writing data. Data types may not
+map the same way in both directions between Trino and the data source. Refer to
+the following sections for type mapping in each direction.
+
+Druid type to Trino type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps Druid types to the corresponding Trino types according to the
+following table:
+
+.. list-table:: Druid type to Trino type mapping
+  :widths: 30, 30, 50
+  :header-rows: 1
+
+  * - Druid type
+    - Trino type
+    - Notes
+  * - ``STRING``
```

Comment:
IIRC we order these in a specific order. Please change to match that order.

Here's the mappings:

### Druid to Trino

`STRING` -> `VARCHAR`
`FLOAT` -> `REAL`
`DOUBLE` -> `DOUBLE`
`LONG` -> `BIGINT` (except the special `__time` column)
`TIMESTAMP` -> `TIMESTAMP` (ONLY the special `__time` column)

One thing which I think is worth calling out is that Druid doesn't have a real `NULL` value for any data type - by default it just treats `NULL` as the default value for a data type (e.g. LONG would be 0, DOUBLE would be 0.0, STRING would be `` (empty string) etc.).

For the reverse direction (Trino to Druid types) consider that Druid is a read-only connector so the scenarios where the write mappings get used are very very limited. It's the exact inverse of the read mappings however.

Reply:
@hashhar and @jhlodin since it is a read-only connector, I assume we don't include the reverse table? Please review again. Thank you.

Reply:
@hashhar We have the line "No other data types are supported", so just to clarify - are those five the only data types supported in Druid?

Reply:
@tlblessing I believe the reverse table is still needed. For those limited cases where the other direction is applicable, again per the "no other data types are supported" caveat, we need to be complete. Otherwise we're suggesting that Druid doesn't support most Trino data types.

Reply:
Yes, only those 5 types are supported. And yes a reverse table is ok to add with the exact same mapping but in reverse.

Reply:
@hashhar and @jhlodin added the reverse mapping table. Ready for another review.

Reply:
>  Druid is a read-only connector so the scenarios where the write mappings get used are very very limited.

@hashhar Follow up question regarding what you said above in this thread. Since the Druid connector is read-only, what are these "very very limited" cases where a Trino to Druid type mapping applies? We want to be complete in the type mapping overview section that we're writing (https://github.com/trinodb/trino/pull/13738) to be sure we're accurate in describing where each mapping applies (i.e. not "read" and "write" mappings, because it's more accurately "read/write" and "DDL" mappings).

If there's no practical uses for a Trino-to-source mapping section in a read-only connector like Druid, @mosabua and I think we should leave it out. We just want to make sure that's accurate.

Reply:
It will get used whenever Trino needs to "write" a query to be sent to Druid. i.e. WHERE clauses, JOIN conditions etc.

But for all practical purposes that mechanism is hidden from the user and there's nothing they can do to affect it so even if you explain the non-write cases where it gets used it doesn't actually help them other than maybe feel like they have some "knowledge".

cc: @ebyhr for opinion

Reply:
It's hidden and also I guess most users think it's part of "read". Documenting "Trino type to {remote} type mapping" in read-only connectors is little confusing in my opinion. Just removing the table looks good to me.

Reply:
Removed read table and adjusted boilerplate language to remove mention of the read direction.

Reply:
Thanks for clarifying .. we will do that for all connectors as applicable.

Question still though .. I don't understand how the type mapping comes into play when "writing" a query for where clauses or joins ... it would only translate Trino SQL into the underlying SQL or whatever language of the data source .. how does type mapping affect that? Are different version created and sent depending on what the mapping is ? Can you think of an example?

Reply:
e.g. With Druid a query like `WHERE __time = TIMESTAMP '2020-01-01 01:01:01'` uses the write mapping to convert the `TIMESTAMP` literal on Trino side to a value to be sent to Druid.

With a query like `SELECT __time FROM druid.default.table` the read mapping is used to convert values read from Druid to Trino values.